introduction/examples/thompson/classroom.py

```diff
@@ -0,0 +1,42 @@
+class Student:
+    def __init__(self, id, fname, lname, courses):
+        self.id = id
+        self.fname = fname
+        self.lname = lname
+        self.courses = courses
+
+
+class Classroom:
+    def __init__(self, capacity):
+        self.capacity = capacity
+        self.students = []
+
+    def add_student(self, student_id):
+        if not isinstance(student_id, int) or len(self.students) >= self.capacity or student_id in self.students:
```

Comment:
Jeżeli sprawdzasz czy argument funkcji jest odpowiedniego typu to radze robić to "osobno" wpierw sprawdzić czy jest typu `int` i jak nie to od razu zwrócić. Dopiero potem sprawdzać poprawność dziedziny danych.